modules/actions/snapshots/sync-snapshots-v3.ts

```diff
@@ -0,0 +1,149 @@
+import { Chain } from '@prisma/client';
+import { prisma } from '../../../prisma/prisma-client';
+import { V3VaultSubgraphClient } from '../../sources/subgraphs';
+import _ from 'lodash';
+import { daysAgo, roundToMidnight } from '../../common/time';
+import { snapshotsV3Transformer } from '../../sources/transformers/snapshots-v3-transformer';
+import { OrderDirection, PoolSnapshot_OrderBy } from '../../sources/subgraphs/balancer-v3-vault/generated/types';
+
+export async function syncSnapshotsV3(
+    vaultSubgraphClient: V3VaultSubgraphClient,
+    chain = 'SEPOLIA' as Chain,
+): Promise<string[]> {
+    // Get the latest snapshot from the DB (assuming there are no gaps)
+    const storedSnapshot = await prisma.prismaPoolSnapshot.findFirst({
+        select: {
+            timestamp: true,
+        },
+        where: {
+            chain,
+        },
+        orderBy: {
+            timestamp: 'desc',
+        },
+    });
+    const storedTimestamp = storedSnapshot?.timestamp || 0;
+
+    // In case there are no snapshots stored in the DB, sync from the subgraph's earliest snapshot
+    let subgraphTimestamp = 0;
+    if (!storedTimestamp) {
+        const { poolSnapshots } = await vaultSubgraphClient.PoolSnapshots({
+            first: 1,
+            orderBy: PoolSnapshot_OrderBy.Timestamp,
+            orderDirection: OrderDirection.Asc,
+        });
+
+        subgraphTimestamp = poolSnapshots[0].timestamp;
+    }
+
+    const timestamp = (storedTimestamp && storedTimestamp + 86400) || subgraphTimestamp;
+
+    console.log(storedTimestamp, subgraphTimestamp, timestamp);
+
+    return syncSnapshotsForADayV3(vaultSubgraphClient, chain, timestamp);
+}
+
+/**
+ * Sync snapshot balances to the database.
+ *
+ * @param vaultSubgraphClient
+ * @param chain
+ * @returns
+ */
+export async function syncSnapshotsForADayV3(
+    vaultSubgraphClient: V3VaultSubgraphClient,
+    chain = 'SEPOLIA' as Chain,
+    timestamp = daysAgo(0), // Current day by default
+): Promise<string[]> {
+    const previous = roundToMidnight(timestamp - 86400); // Previous day stored in the DB
+    const next = roundToMidnight(timestamp); // Day to fetch snapshots for
+
+    // TODO: do we want to have a bucket for the current day?
```

Comment:
I assume "next" would be the current day?

Reply:
it's tricky, because current is not today, and i find current to be confusing with today.

Reply:
In normal operation, when everything is synced, current is today. Isnt it?